lfs/client.go

```diff
@@ -125,8 +126,15 @@ func Download(oid string) (io.ReadCloser, int64, error) {
 		return nil, 0, err
 	}
 	LogTransfer("lfs.data.download", res)
+	
+	buf, _ := ioutil.ReadAll(res.Body)
+	body := myReader{bytes.NewBuffer(buf)}	
```

Comment:
The issue is that multiple threads call `Download()`. If the thread 1 makes a request, and then thread 2 makes a request before thread 1 closes the stream, then the `*Http.Client` will spin up a new connection because it can't reuse the old one. NTLM is connection-oriented, so we will have to redo the 3-5 hop auth cycle.

If we instead close/copy the stream before returning it to the caller, every caller will reuse the connection and we only have to do the auth negotiation once.

I agree that this doesn't make sense for non-NTLM scenarios, so I can wrap it in an `if` block to only dispose early in NTLM cases. Long term, we may want a pool of `*http.Client`s and then we can just pick one that has an available connection.